bin/cylc-check-software

```diff
@@ -183,21 +185,26 @@ def cmd_find_ver(module, min_ver, cmd_base, ver_opt, ver_extr, outfile=1):
        version-checking option(s) & version-extraction regex."""
     msg = '%s (%s)' % (module, string_ver(min_ver) + '+' if
                        min_ver is not None else 'none')
-    if Popen(['which', cmd_base], stdout=PIPE, stderr=PIPE).wait():
-        res = [NOTFOUND_MSG, False]
-    else:
-        try:
-            output = Popen([cmd_base, ver_opt], stdout=PIPE,
-                           stderr=PIPE).communicate()[outfile - 1].strip()
-            version = re.search(ver_extr, output).groups()[0]
-            if min_ver is None:
-                res = ['%s (%s)' % (FOUND_NOVER_MSG, version), True]
-            elif parse_version(version) >= min_ver:
-                res = ['%s (%s)' % (MINVER_MET_MSG, version), True]
-            else:
-                res = ['%s (%s)' % (MINVER_NOTMET_MSG, version), False]
-        except AttributeError:
-            res = [FOUND_UNKNOWNVER_MSG, False]
+    for cmd in cmd_base:
+        try_next_cmd = True
+        if Popen(['which', cmd], stdout=PIPE, stderr=PIPE).wait():
+            res = [NOTFOUND_MSG, False]
+        else:
+            try:
+                output = Popen([cmd, ver_opt], stdout=PIPE,
+                               stderr=PIPE).communicate()[outfile - 1].strip()
```

Comment:
Note we now set `stdin=open(os.devnull)` in `Popen` calls where possible (see other examples under `bin/`

Reply:
Sensible. I'll get those changed.